File: app/src/main/java/nejati/me/service/generator/ComponentService.java
package nejati.me.service.generator;


import dagger.Component;
import nejati.me.service.di.component.NetComponent;
import nejati.me.service.scope.CustomScope;


@CustomScope
@Component(dependencies = NetComponent.class)
public interface ComponentService {
    void inject(SingletonService singletonService);
}

File: app/src/main/java/nejati/me/service/generator/ComponentService.kt
package nejati.me.service.generator


import dagger.Component
import nejati.me.service.di.component.NetComponent
import nejati.me.service.scope.CustomScope


@CustomScope
@Component(dependencies = [NetComponent::class])
interface ComponentService {
    /**
     * @param singletonService
     */
    fun inject(singletonService: SingletonService)
}
